### PR TITLE
Use Embeddings in pinecone

### PR DESCRIPTION
**Pull request by baskaryan**
cc @eyurtsev @olivier-lacroix @jamescalam 

redo of #2741

### CODE REVIEW

**Comment by baskaryan** [comment]
technically a little breaking if someone had been passing in `embedding_function` as named param

File: libs/langchain/langchain/vectorstores/pinecone.py [hunk 1]
```diff
@@ -56,16 +57,28 @@ def __init__(
...
+            )
         self._index = index
-        self._embedding_function = embedding_function
+        self._embedding = embedding
         self._text_key = text_key
         self._namespace = namespace
         self.distance_strategy = distance_strategy
 
-    @property
-    def embeddings(self) -> Optional[Embeddings]:
```
Comment (eyurtsev) [inline libs/langchain/langchain/vectorstores/pinecone.py:66]: should we have it return _embedding when possible?

**Review by eyurtsev — APPROVED** [review]

File: libs/langchain/langchain/vectorstores/pinecone.py (hunk 1 above)
Comment (baskaryan) [reply libs/langchain/langchain/vectorstores/pinecone.py:66]: actually not sure what this is for, cc @hinthornw

Comment (eyurtsev) [reply libs/langchain/langchain/vectorstores/pinecone.py:66]: I think tracing related?

**Comment by olivier-lacroix** [comment]
Great, thanks @baskaryan !!

**Pull request merged**